plugins/optimization-detective/storage/data.php

```diff
@@ -150,14 +150,15 @@ function od_get_url_metrics_slug( array $query_vars ): string {
  *
  * @see od_verify_url_metrics_storage_hmac()
  * @see od_get_url_metrics_slug()
+ * @todo This should also include an ETag as a parameter. See <https://github.com/WordPress/performance/issues/1466>.
  *
- * @param string $slug Slug (hash of normalized query vars).
- * @param string $url  URL.
- *
+ * @param string   $slug                Slug (hash of normalized query vars).
+ * @param string   $url                 URL.
+ * @param int|null $cache_purge_post_id Cache purge post ID.
  * @return string HMAC.
  */
-function od_get_url_metrics_storage_hmac( string $slug, string $url ): string {
-	$action = "store_url_metric:$slug:$url";
+function od_get_url_metrics_storage_hmac( string $slug, string $url, ?int $cache_purge_post_id = null ): string {
```

Comment:
See above.

Also, should this maybe be a _required_ parameter? That would emphasize that it's needed to optimize responses anyway, as you already check above.

Reply:
> Also, should this maybe be a _required_ parameter? That would emphasize that it's needed to optimize responses anyway, as you already check above.

The check I added means that _by default_ it will not offer to optimize responses when `od_get_cache_purge_post_id()` returns `null`. But this can be overridden with the `od_can_optimize_response` filter. So we're not guaranteed that it will always be not-`null`.

Reply:
What would be a use-case for that? I'm thinking it might be better to _not_ make that possible. For example, we could make it so that the filter only is applied when there is a post ID. That feels more robust to me.

Reply:
Sometimes I've made sites where I use author archive pages to create team bio pages, for example. In such a case, there often isn't any blog post actually authored by that user. Nevertheless, if I go to their author page it doesn't actually return a 404. It serves a page with the author's name in the document title and if the theme has an `author.php` template which is configured to display the user bio even when there are no posts, then it will work as intended. For example, the Twenty Ten theme shows the user bio on `author.php` even when there are no posts by that author:

![image](https://github.com/user-attachments/assets/f0a17910-c6bd-4753-b3f2-8f145ae5bc8a)